src/Acuminator/Acuminator.Utilities/Roslyn/PXFieldAttributes/AttributeInformation.cs

```diff
@@ -141,26 +144,49 @@ public bool IsAttributeDerivedFromClass(ITypeSymbol attributeType, ITypeSymbol t
 		/// <summary>
 		/// Query if Acumatica attribute is bound.
 		/// </summary>
-		/// <param name="attributeType">Type of the attribute.</param>
+		/// <param name="attribute">Data of the attribute.</param>
 		/// <returns/>
-		public bool IsBoundAttribute(ITypeSymbol attributeType)
+		public BoundType IsBoundAttribute(AttributeData attribute)
 		{
-			attributeType.ThrowOnNull(nameof(attributeType));
+			attribute.ThrowOnNull(nameof(attribute));
 
-			if (!attributeType.InheritsFromOrEquals(_eventSubscriberAttribute))
-				return false;
+			if (!attribute.AttributeClass.InheritsFromOrEquals(_eventSubscriberAttribute))
+				return BoundType.Unknown;
 
-			return BoundBaseTypes.Any(boundBaseType => IsAttributeDerivedFromClassInternal(attributeType, boundBaseType));
+			if (BoundBaseTypes.Any(boundBaseType => IsAttributeDerivedFromClassInternal(attribute.AttributeClass, boundBaseType)))
+				return BoundType.DbBound;
+
+			if (attribute.AttributeClass.GetMembers().Select(a => a.Name).Contains(IsDBField))
+			{
+				foreach (var argument in attribute.NamedArguments)
+				{
+					if (argument.Key == IsDBField)
+					{
+						if (!argument.Value.IsNull && argument.Value.Value is bool boolValue && boolValue == true)
+							return BoundType.DbBound;
+						else
+							return BoundType.Unbound;
+					}
+				}
+				return BoundType.Unknown;
+			}
+			return BoundType.Unbound;
 		}
 
 		/// <summary>
 		/// Query if collection of attributes contains bound attribute.
 		/// </summary>
-		/// <param name="attributeTypes">The attributes collection.</param>
+		/// <param name="attributes">The attributes collection.</param>
 		/// <returns/>
-		public bool ContainsBoundAttributes(IEnumerable<ITypeSymbol> attributeTypes)
+		public BoundType ContainsBoundAttributes(IEnumerable<AttributeData> attributes)
 		{
-			return attributeTypes.Any(IsBoundAttribute);
+			foreach (var attribute in attributes)
+			{
+				BoundType result = IsBoundAttribute(attribute);
+				if (result == BoundType.DbBound || result == BoundType.Unknown)
```

Comment:
Why do we need to return in case if `result` is `BoundType.Unknown`?
For this case we'd like to get `BoundType.Bound`:
```C#
[PXDependsOnFields(typeof(amount), typeof(price))] // doesn't inherit from PXEventSubscriberAttribute, so BoundType will be Unknown
[PXDBDecimal] // BoundType.DbBound
public decimal? Total => Amount * Price;
```

Reply:
If one attribute has `Unknown` type, all field is `Unbound`? Or return `Unknown`?

Reply:
@saloponov I suggest adding a new value for the enum: `NotDefined`
Return `NotDefined` for all attributes that are not inherited from `PXEventSubscriberAttribute`, `Unknown` for attributes with unknown `IsDBField` property, and `Bound` / `Unbound` as it is now.